### PR TITLE
Added PasswordResetView.

### PR DESCRIPTION
**Pull request by normanjaeckel**
@FinnStutzenstein This is a WIP.

Can you please add a first view where a user who as forgotten his password can insert his email address. This view should send a POST request to my url. Then I can test it and try to get out a proper email.

Thanks.

### CODE REVIEW

**Comment by normanjaeckel** [comment]
Ping.

**Comment by FinnStutzenstein** [comment]
You can test now with the client. If the server gives a meaningful response, I can finish the client.
You do not need anything for the one-time url in the client, right?

Does the confirm-view needs a `csrf-exempt` or did django include this already?

**Comment by normanjaeckel** [comment]
@FinnStutzenstein Thanks for building the view. Now you can click on your link and receive the email (test this with command line backend). Next step is that the client should provide a form on this link and send a POST request to my second view. Please ask if there are any questions.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel The view works. I do have two todos (related to the same topic): Does the server send a success/failure message in both new views? Currently not (which is ok for the success case), but can be done.

Do you want to have error messages in the reset-view? I think it is not good to say 'this email does not exist.'.

Error messages in the confirm view? Should come from the server I think.

Notice: I rebased the branch on the current master.

**Comment by normanjaeckel** [comment]
All done and rebased. Please test.

**Comment by normanjaeckel** [comment]
Travis failed in client tests (maybe because of my changes that create requests return the id now).

File: openslides/users/views.py [hunk 1]
```diff
@@ -524,3 +528,113 @@ def post(self, request, *args, **kwargs):
...
+        """
+        return textwrap.dedent(
+            """
+            You're receiving this email because you requested a password reset for your user account at {site_name}.
+
+            Please go to the following page and choose a new password:
+
+            {protocol}://{domain}{path}?user_id={user_id}&token={token}
+
+            Your username, in case you've forgotten: {username}
```
Comment (tsiegleauq) [inline openslides/users/views.py]: including the username in the E-Mail is a security flaw.
Intercepting the E-Mail would give you full access.
Knowing the E-Mail address of a registered administrator is sufficient to take over the software.

File: client/src/app/site/login/components/reset-password-confirm/reset-password-confirm.component.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                setTimeout(() => {
+                    this.matSnackBar.open('');
+                    this.matSnackBar.open(
+                        this.translate.instant('The link is broken. Please contact your system administrator.'),
+                        this.translate.instant('OK'),
+                        {
+                            duration: 0
+                        }
+                    );
+                    this.router.navigate(['/']);
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/login/components/reset-password-confirm/reset-password-confirm.component.ts]: Does the snackbar closes automatically? I am redirected and the snackbar stays in place until I click OK.

File: openslides/users/views.py (hunk 1 above)
Comment (FinnStutzenstein) [reply openslides/users/views.py]: @normanjaeckel Ping

Comment (normanjaeckel) [reply openslides/users/views.py]: This is the django default text. If someone lost his username he cannot reset his account. @emanuelschuetze What do you think? (It wouldn't be a problem for me if you change the text.

Comment (tsiegleauq) [reply openslides/users/views.py]: Since it's possible to enter an email twice that should be OK

**Pull request merged**